components/support/utils/src/main/java/mozilla/components/support/utils/URLStringUtils.kt

```diff
@@ -28,4 +28,10 @@ object URLStringUtils {
         }
         return uri.toString()
     }
+
+    private val isURLLenient by lazy {
+        // Be lenient about what is classified as potentially a URL. Anything that contains a :
+        // or . and has no internal spaces is potentially a URL.
+        "^\\s*\\w+(://|:|\\.)\\w+\\S*\\s*$".toRegex()
```

Comment:
I don't think `\w` matches unicode characters so http://faß.de would not be considered valid. It [sounds like](https://www.regular-expressions.info/wordboundaries.html) Java might support unicode for `\b`.

Can we make sure there are unicode tests in the test file (if there aren't already)?

Reply:
Good point!